### ISSUE TITLE
possible bug with emitter.end

### ISSUE THREAD

**Issue by nostalgicdodo** [rpominov/kefir#122]
I have an AJAX helper function as follows,
```js
function fetch (url, payload) {
	return kefir.stream(function (emitter) {

		var jqXHR = $.ajax({
			url: url,
			data: payload || {}
		});
		jqXHR.done(function (data, textStatus, jqXHR) {
			emitter.emit(data);
			emitter.end();	//- this invocation triggers the error
		});
		jqXHR.fail(function (jqXHR, textStatus, errorThrown) {
			emitter.error(textStatus);
			emitter.end();
		});

		return function () {
			jqXHR.abort();
			emitter.end();
		};

	});
}
```
When this function is used, I get the following message on the console,
```
Uncaught TypeError: Cannot read property 'cleanup' of null
_clear @ kefir-v2.7.1.js?ver=20150709:613
_clear @ kefir-v2.7.1.js?ver=20150709:1511
_emitEnd @ kefir-v2.7.1.js?ver=20150709:644
end @ kefir-v2.7.1.js?ver=20150709:1429
(anonymous function) @ search_products_async.js?ver=20150709:17
```

This occurs in v2.7.1 but not in v2.7.0. When the `emitter.end()` statement is commented out, everything works just fine.

The app functionality doesn't break; only this message pops up.

**Comment by rpominov** [comment]
Thanks! Also found another bug when was fixing this, although the fix was the same :smile: 
Can you check if it works for you (it's in master)?

**Comment by nostalgicdodo** [comment]
yes, it's working fine now. thanks!

**Comment by rpominov** [comment]
:+1: I've released `2.7.2` with this fix.
